### ISSUE TITLE
vaadin.whitelisted-packages should have a sensible default

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/flow#16877]
### Description of the bug

When downloading a project from start.spring.io, it does not have the `vaadin.whitelisted-packages` property defined. 

Building a native image out of the app and running it fails.

### Expected behavior

The app works out of the box also when compiled to native. 

We should be able to produce the default value `com.vaadin,org.vaadin,dev.hilla,com.example.application` easily as a fallback as the first packages are our own, and the last might be determined by finding where the application class is located. 

### Minimal reproducible example

- Download an app with the Vaadin dependency from start.spring.io. 
- Native compile it with `mvn -Pnative -Pproduction native:compile`
- Run the created binary 

### Versions

- Vaadin / Flow version: 24.0
- Java version: 17, GraalVM
- OS version: Mac

**Comment by tarekoraby** [comment]
If start.spring.io is the main concern, then we should probably add com.example.demo.

**Comment by marcushellberg** [comment]
People can change that. Could we add the package that contains the main spring boot application instead? That should most often be correct.

**Comment by tarekoraby** [comment]
Sounds reasonable to me. Though that might not be the desired setup in some of the larger projects, this solution would ensure that most new evaluators have setup that works.

**Comment by marcushellberg** [comment]
Agreed. If there's an explicit value defined as a property, that should override. But this way, most (all?) projects from both start.vaadin.com and start.spring.io would work out of the box

**Comment by mshabarov** [comment]
Needs to be reproduced and see what's the original error we have when building with native profile.

**Comment by mcollovati** [comment]
Projects created with start.vaadin.com currently have `vaadin.whitelisted-packages` set.
I tried to remove it, but I cannot see any error; the application works out of the box

**Comment by mcollovati** [comment]
Trying to run the executable produced by the skeleton project fails, but the error seems more related to PWA and AWT than the `whitelisted-packages` setting.
If I remove the PWA annotation, the native executable works fine.

@marcushellberg any chance that you remember the exact error you got? Or could you reproduce it again and post the log with the evidence of the error?

**Comment by mcollovati** [comment]
Closing this as obsolete.

The native image failure was not caused by the missing `vaadin.allowed-packages` property. As noted earlier in this issue, the real problem was the `@PWA` annotation, which loaded AWT at runtime.
That was fixed in #20516 (Fixes #19497), where PWA icons are generated at build time instead. The fix is also backported to 24.x.

Also, `vaadin.allowed-packages` has no effect on class scanning in production mode, so it cannot break a native build. In production, Flow scans the package of the `@SpringBootApplication` class, taken from Spring's `AutoConfigurationPackages`. So the application package is already detected automatically, which is what this issue asked for.

Adding a default value would make things worse: it would limit dev mode scanning to those packages and break projects that use other package names.
